### PR TITLE
Update to apollo client 0.4.0

### PR DESCRIPTION
**Pull request by stubailo**
A few hacks we should fixe before merging

### CODE REVIEW

File: src/connect.tsx [hunk 1]
```diff
@@ -254,33 +264,31 @@ export default function connect(opts?: ConnectOptions) {
...
-        if (isEqual(oldQueries, queryHandles)) {
+        if (isEqual(oldQueries, queryOptions)) {
           return false;
         } else if (oldQueries) {
           // unsubscribe from previous queries
           this.unsubcribeAllQueries();
         }
 
-        if (isObject(queryHandles) && Object.keys(queryHandles).length) {
-          this.queryHandles = queryHandles;
```
Comment (stubailo) [inline src/connect.tsx]: As far as I know, this line was not correct - it was assigning query options to the same object we use to hold the observables/handles/subscriptions.

File: typings/modules/apollo-client/index.d.ts [hunk 2]
```diff
@@ -274,12 +274,13 @@ import { Observable, Observer, Subscription } from '~apollo-client/util/Observab
 export class ObservableQuery extends Observable<GraphQLResult> {
     subscribe(observer: Observer<GraphQLResult>): QuerySubscription;
     result(): Promise<GraphQLResult>;
-}
-export interface QuerySubscription extends Subscription {
```
Comment (stubailo) [inline typings/modules/apollo-client/index.d.ts]: I couldn't figure out how to get `typings` to do the right thing here, so I edited manually to get things to pass. How do we regenerate the types here?

**Comment by jbaxleyiii** [comment]
@stubailo thank you so much for doing this!

**Comment by stubailo** [comment]
@jbaxleyiii I think it's ready to merge as soon as you have time to look it over!

**Pull request merged**